docs/integrations/databases/neon.mdx

```diff
@@ -119,72 +104,100 @@ export const db = drizzle(sql, {
 
-1. At the root of your project, create a `drizzle.config.ts` and add the following configuration:
-    ```typescript {{ filename: 'drizzle.config.ts' }}
-    import { defineConfig } from "drizzle-kit";
-    import { loadEnvConfig } from "@next/env";
+```typescript {{ filename: 'drizzle.config.ts' }}
+import { defineConfig } from "drizzle-kit";
+import { loadEnvConfig } from "@next/env";
 
-    loadEnvConfig(process.cwd());
+loadEnvConfig(process.cwd());
 
-    if (!process.env.DATABASE_URL) {
-      throw new Error("DATABASE_URL must be a Neon postgres connection string");
-    }
+if (!process.env.DATABASE_URL) {
+  throw new Error("DATABASE_URL must be a Neon postgres connection string");
+}
 
-    export default defineConfig({
-      dialect: "postgresql",
-      dbCredentials: {
-        url: process.env.DATABASE_URL,
-      },
-      schema: "./app/db/schema.ts",
-      out: "./migrations",
-    });
-    ```
-1. Generate and run the database migrations by running the following commands:
-    ```sh {{ filename: 'terminal' }}
-    npx drizzle-kit generate
-    npx drizzle-kit migrate
-    ```
+export default defineConfig({
+  dialect: "postgresql",
+  dbCredentials: {
+    url: process.env.DATABASE_URL,
+  },
+  schema: "./app/db/schema.ts",
+});
+```
+
+Run the following command to push the schema to the database:
+```sh {{ filename: 'terminal' }}
+npx drizzle-kit push
+```
 
-### Create the UI of the home page
+### Creating server actions with Clerk
 
-Add the following code to the `app/page.tsx` file to create the UI of the home page:
+To add and delete user messages from the database, create an `actions.ts` file in the `app` directory. Use the `currentUser` function from the Clerk SDK to get the user ID and interact with the database.
+
+```typescript {{ filename: 'app/actions.ts' }}
+"use server";
+
+import { currentUser } from "@clerk/nextjs/server";
+import { UserMessages } from "./db/schema";
+import { db } from "./db";
+import { eq } from "drizzle-orm";
+
+export async function createUserMessage(formData: FormData) {
+  const user = await currentUser();
+  if (!user) throw new Error("User not found");
+
+  const message = formData.get("message") as string;
+  await db.insert(UserMessages).values({
+    user_id: user.id,
+    message,
+  });
+}
+
+export async function deleteUserMessage() {
+  const user = await currentUser();
+  if (!user) throw new Error("User not found");
+
+  await db.delete(UserMessages).where(eq(UserMessages.user_id, user.id));
+}
+```
+
+### Render the UI based on the user ID
+
+To conditionally render the page based on the user ID, create a `page.tsx` file in the `app` directory. Use the `currentUser` function from the Clerk to get the user ID and query the database for existing messages.
+
+If a message exists, display the message and a button to delete it. If no message exists, display a form to create a new message.
```

Comment:
There will already be a `page.tsx` file in the `app` directory - this tutorial requires the user have a Next.js app set up and this page is present out-of-the-box. So we can remove this instruction.

We don't use `currentUser()` to conditionally render the page - clerkMiddleware() will protect the page for us. We use `currentUser()` to get the `user.id` in order to query the database - the `if (!user)` check is only there as a 
TypeScript check for checking if `user` is null.

Also, almost always provide a reference link where you can. I've added it to `currentUser()`:

```suggestion
### Create the UI of the home page

In your `app/pase.tsx` file, add the following code to create the UI of the home page. If there is an existing message, the user can view and delete it; otherwise, they can add a new message.

To get the user's messages, Clerk's [`currentUser()`](/docs/references/nextjs/current-user) helper is used to get ID of the current user of your application. Then, the ID is used to query the database for the user's messages.

To allow the user to delete or add a message, the forms are handled by the `deleteUserMessage()` and `createUserMessage()` actions that you created in the previous step.
```

Reply:
Great suggestion. I edited lightly to use active voice in the second paragraph.

```suggestion
### Create the UI for the Home Page

In your `app/page.tsx` file, add the following code to create the UI for the home page. If a message exists, the user can view and delete it; otherwise, they can add a new message.

To retrieve the user's messages, use Clerk's [`currentUser()`](/docs/references/nextjs/current-user) to obtain the current user's ID. Then, use this ID to query the database for the user's messages.

To enable the user to delete or add a message, use the `deleteUserMessage()` and `createUserMessage()` actions created in the previous step.
```